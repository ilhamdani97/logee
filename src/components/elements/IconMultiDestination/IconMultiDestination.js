import React from 'react';
import PropTypes from 'prop-types';
import compareProps from '../../../utils/compareProps';

const marker = {
  '0': `M12 8C9.76 8 8 9.76 8 12C8 14.24 9.76 16 12 16C14.24 16 16 14.24 16 12C16 9.76 14.24 8 12 8Z`,
  '1': `M13 16.7308C13 16.8026 12.9713 16.8654 12.914 16.9192C12.8566 16.9731 12.7897 17 12.7133 17H11.3082C11.2318 
    17 11.1649 16.9731 11.1075 16.9192C11.0502 16.8654 11.0215 16.8026 11.0215 
    16.7308V8.88425L9.32975 9.42261C9.28196 9.44056 9.22461 9.44953 9.15771 
    9.44953C9.05257 9.44953 9 9.39569 9 9.28802V7.9825C9 7.91072 9.03345 
    7.86137 9.10036 7.83445L11.4516 7.05384C11.4994 7.03589 11.5472 7.02243 
    11.595 7.01346C11.6428 7.00449 11.7049 7 11.7814 7H12.7276C12.9092 7 13 
    7.08524 13 7.25572V16.7308Z`,
  '2': `M15.608 9.562C15.608 10.15 15.4587 10.654 15.16 11.074C14.8707 11.4847 14.4367 11.886 13.858 12.278L12.598 13.118C11.9447 
    13.5567 11.45 13.9813 11.114 14.392C10.778 14.7933 10.61 15.2833 10.61 
    15.862V15.918H15.216C15.2907 15.918 15.356 15.9413 15.412 15.988C15.468 
    16.0347 15.496 16.0953 15.496 16.17V17.234C15.496 17.3087 15.468 17.374 
    15.412 17.43C15.356 17.486 15.2907 17.514 15.216 17.514H9.28C9.20533 17.514 
    9.14 17.486 9.084 17.43C9.028 17.374 9 17.3087 9 17.234V15.862C9 15.4793 9.056 
    15.1107 9.168 14.756C9.28 14.392 9.434 14.056 9.63 13.748C9.826 13.4307 10.05 13.1413 
    10.302 12.88C10.5633 12.6187 10.8387 12.39 11.128 12.194L12.514 11.27C12.7847 11.0927 
    13.004 10.9387 13.172 10.808C13.34 10.6773 13.4707 10.556 13.564 10.444C13.6573 10.3227 
    13.718 10.1967 13.746 10.066C13.7833 9.926 13.802 9.758 13.802 9.562C13.802 9.17933 13.676 
    8.91333 13.424 8.764C13.172 8.60533 12.752 8.526 12.164 8.526C11.9587 8.526 11.7347 8.53533 
    11.492 8.554C11.2587 8.56333 11.016 8.57733 10.764 8.596C10.5213 8.61467 10.2787 8.638 10.036 
    8.666C9.80267 8.68467 9.57867 8.708 9.364 8.736H9.336C9.26133 8.736 9.20067 8.708 9.154 
    8.652C9.10733 8.596 9.084 8.53067 9.084 8.456V7.616C9.084 7.54133 9.10733 7.48533 9.154 
    7.448C9.21 7.40133 9.27533 7.36867 9.35 7.35C9.53667 7.30333 9.742 7.26133 9.966 
    7.224C10.1993 7.17733 10.4373 7.14 10.68 7.112C10.932 7.07467 11.184 7.04667 11.436 
    7.028C11.6973 7.00933 11.94 7 12.164 7C13.3867 7 14.264 7.21 14.796 7.63C15.3373 8.05 
    15.608 8.694 15.608 9.562Z`,
  '3': `M15.468 14.602C15.468 15.1247 15.3793 15.5773 15.202 15.96C15.0247 16.3333 14.7773 16.646 14.46 16.898C14.1427 17.1407 13.7693 
    17.3227 13.34 17.444C12.9107 17.5653 12.4487 17.626 11.954 17.626C11.4313 17.626 10.9507 
    17.5933 10.512 17.528C10.0827 17.4627 9.66733 17.3787 9.266 17.276C9.19133 17.2573 9.126 
    17.2293 9.07 17.192C9.02333 17.1453 9 17.0847 9 17.01V16.142C9 16.0673 9.02333 16.0067 9.07 
    15.96C9.11667 15.9133 9.17733 15.89 9.252 15.89H9.28C9.84 15.946 10.3393 15.9927 10.778 
    16.03C11.226 16.0673 11.618 16.086 11.954 16.086C12.5233 16.086 12.9247 15.9647 13.158 
    15.722C13.4007 15.47 13.522 15.0967 13.522 14.602V14.308C13.522 14.1213 13.5033 13.944 
    13.466 13.776C13.438 13.608 13.3727 13.4587 13.27 13.328C13.1767 13.1973 13.046 13.0947 
    12.878 13.02C12.7193 12.936 12.5047 12.894 12.234 12.894H9.756C9.57867 12.894 9.49 12.8053 
    9.49 12.628V11.746C9.49 11.5687 9.57867 11.48 9.756 11.48H12.234C12.5047 11.48 12.7193 
    11.4427 12.878 11.368C13.046 11.2933 13.1767 11.1953 13.27 11.074C13.3727 10.9433 13.438 
    10.7987 13.466 10.64C13.5033 10.472 13.522 10.2947 13.522 10.108V9.884C13.522 9.70667 13.4987 
    9.53867 13.452 9.38C13.4147 9.212 13.3353 9.06733 13.214 8.946C13.102 8.82467 12.9433 8.72667 
    12.738 8.652C12.542 8.57733 12.2807 8.54 11.954 8.54C11.702 8.54 11.478 8.54467 11.282 
    8.554C11.0953 8.56333 10.904 8.57733 10.708 8.596C10.512 8.60533 10.302 8.624 10.078 
    8.652C9.854 8.68 9.588 8.708 9.28 8.736H9.252C9.17733 8.736 9.11667 8.71267 9.07 
    8.666C9.02333 8.61933 9 8.55867 9 8.484V7.616C9 7.54133 9.02333 7.48533 9.07 7.448C9.126 
    7.40133 9.19133 7.36867 9.266 7.35C9.66733 7.24733 10.0827 7.16333 10.512 7.098C10.9507 
    7.03267 11.4313 7 11.954 7C12.458 7 12.9247 7.05133 13.354 7.154C13.7833 7.25667 14.152 
    7.42467 14.46 7.658C14.7773 7.882 15.0247 8.176 15.202 8.54C15.3793 8.904 15.468 9.352 15.468 
    9.884V10.108C15.468 10.6027 15.3607 11.0227 15.146 11.368C14.9407 11.704 14.6467 11.9747 
    14.264 12.18C14.656 12.3947 14.9547 12.6933 15.16 13.076C15.3653 13.4493 15.468 13.86 
    15.468 14.308V14.602Z`
};

function IconMultiDestination({ number }) {
  const fill = number === 0 ? '#F76707' : '#FFE8CC';

  return (
    <svg fill="none" height="24" viewBox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg">
      <path
        clipRule="evenodd"
        d="M12 2C6.4 2 2 6.4 2 12C2 17.6 6.4 22 12 22C17.6 22 22 17.6 22 12C22 6.4 17.6 2 12 2Z"
        fill="#E8590C"
        fillRule="evenodd"
      />
      <path d={marker[number]} fill={fill}/>
    </svg>
  );
}

export default React.memo(IconMultiDestination, compareProps);

IconMultiDestination.defaultProps = {
  number: 0
};

IconMultiDestination.propTypes = {
  number: PropTypes.number
};

export const IconMultiDestinationString = number => {
  const fill = number === 0 ? '#F76707' : '#FFE8CC';

  return (`
    <svg fill="none" height="24" viewBox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg">
      <path
        clipRule="evenodd"
        d="M12 2C6.4 2 2 6.4 2 12C2 17.6 6.4 22 12 22C17.6 22 22 17.6 22 12C22 6.4 17.6 2 12 2Z"
        fill="#E8590C"
        fillRule="evenodd"
      />
      <path d="${marker[number]}" fill="${fill}"/>
    </svg>
  `);
};
