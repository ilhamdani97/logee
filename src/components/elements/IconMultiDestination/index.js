export { default } from './IconMultiDestination';

export { IconMultiDestinationString } from './IconMultiDestination';
