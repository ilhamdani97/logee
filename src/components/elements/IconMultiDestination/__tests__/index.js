import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import IconMultiDestination, { IconMultiDestinationString } from '../IconMultiDestination';

const { defaultProps } = IconMultiDestination.type;

describe('src/components/elements/IconMultiDestination', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<IconMultiDestination {...defaultProps} />);
    expect(tree).toMatchSnapshot();
  });

  test('render string', () => {
    const icon = IconMultiDestinationString(0);
    expect(icon).toMatchSnapshot();
  });
});
