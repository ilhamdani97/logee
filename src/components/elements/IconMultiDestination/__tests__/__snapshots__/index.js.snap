// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/IconMultiDestination render 1`] = `
<svg
  fill="none"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    clipRule="evenodd"
    d="M12 2C6.4 2 2 6.4 2 12C2 17.6 6.4 22 12 22C17.6 22 22 17.6 22 12C22 6.4 17.6 2 12 2Z"
    fill="#E8590C"
    fillRule="evenodd"
  />
  <path
    d="M12 8C9.76 8 8 9.76 8 12C8 14.24 9.76 16 12 16C14.24 16 16 14.24 16 12C16 9.76 14.24 8 12 8Z"
    fill="#F76707"
  />
</svg>
`;

exports[`src/components/elements/IconMultiDestination render string 1`] = `
"
    <svg fill=\\"none\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" width=\\"24\\" xmlns=\\"http://www.w3.org/2000/svg\\">
      <path
        clipRule=\\"evenodd\\"
        d=\\"M12 2C6.4 2 2 6.4 2 12C2 17.6 6.4 22 12 22C17.6 22 22 17.6 22 12C22 6.4 17.6 2 12 2Z\\"
        fill=\\"#E8590C\\"
        fillRule=\\"evenodd\\"
      />
      <path d=\\"M12 8C9.76 8 8 9.76 8 12C8 14.24 9.76 16 12 16C14.24 16 16 14.24 16 12C16 9.76 14.24 8 12 8Z\\" fill=\\"#F76707\\"/>
    </svg>
  "
`;
